app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,15 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  
+  def facebook
+   # You need to implement the method below in your model (e.g. app/models/user.rb)
+   @user = User.from_omniauth(request.env["omniauth.auth"])
+
+   if @user.persisted?
+     sign_in_and_redirect @user, :event => :authentication #this will throw if @user is not activated
+     set_flash_message(:notice, :success, :kind => "Facebook") if is_navigational_format?
```

Comment:
Style/HashSyntax: Use the new Ruby 1.9 hash syntax. (https://github.com/bbatsov/ruby-style-guide#hash-literals)